R/printFFTrees_function.R

```diff
@@ -30,8 +38,24 @@ print.FFTrees <- function(x = NULL,
 
   n.cues <- x$trees$definitions$nodes[tree]
 
+  # - Validate data argument: ----
+
+  data <- tolower(data)  # for robustness
+
+  if (!data %in% c("test", "train")){ # unknown value:
```

Comment:
I would recommend that if the user specifies a different value, the function `stop()` and throw an error. If someone explicitly gave a new value of `data` other than `"train"` or `"test"`, I think they should be forced to correct their spelling.

Reply:
Agreed (and changed in new commit). 

btw: The message and recourse to _training_ data when no "test" data exists is modeled on `plot.FFTrees()`.  If we wanted to enforce more careful user inputs, we may consider replacing both `message()`s by corresponding `warning()`s.  But as `data = "test"` per default, this would result in many warnings whenever no test data was specified.  Hence, I'd hope that messages — in combination with the explicit labels in the output — are sufficient here.

Reply:
Good to know. Indeed my opinions on accommodating users has changed dramatically since 5 years ago when the package was first written. I'm sure there are lots of places in the package now where I'd now prefer swapping out messages and warnings and re-assigning 'bad' inputs for strict but informative errors!

Reply:
That's interesting, and I see the point in favor of informative errors.  But since a **print** method is such an integral thing for an object I'd like to avoid inevitable errors with it when `FFTrees` objects have no test data.  Hence, I think the warning approach makes sense, as they're quite noticeable in the output. (The alternative of printing _training_ data per default would avoid errors, but clash with the current **plotting** method and our general emphasis on using models to predict, rather than fit data...)